### DISCUSSION TITLE
WebRadio images broken when stream URL contains question mark

### DISCUSSION THREAD

**Discussion by MrsTrellis** [rern/rAudio#331] [Fixed Issues]
For some time (over a year), whenever I add a coverart image to a WebRadio stream containing a question mark (?), the images are broken, eg:
http://tx.planetradio.co.uk/icecast.php?i=countryhits.aac

It looks like the image gets uploaded successfully and the thumb created, so the issue is in the display on the rAudio GUI.

mcfly:/srv/http/data/webradiosimg  #  ls -al | grep country
-rw-r--r--  1 root root  2968 May 18 07:55 http:||tx.planetradio.co.uk|icecast.php?i=countryhits.aac-thumb.jpg
-rw-r--r--  1 root root 75464 May 18 07:55 http:||tx.planetradio.co.uk|icecast.php?i=countryhits.aac.jpg

![image](https://user-images.githubusercontent.com/77433570/118640106-835e2d80-b7d0-11eb-883b-5b0c67338a5a.png)

**Comment by AndyFx12** [comment]
Hi MrsTrellis,
it´s a known issue - in most cases, URL´s with "?" will be resolved to a static URL. In your example URL, result is
`http://edge-baueral-03-gos2.sharp-stream.com/countryhits.aac`
In this result-URL there´s no question mark ...

Not perfect - but at the moment you have to deal with it.

Some tip, if you have an Android Device: RadioDroid2 is a perfect "add-on"-app for exploring webradios. It´s able to play stations with MPD - and rAudio1 can save unknown stations at context-menu of playlist-item. Questions-marks will be removed before saving ...

regards, Andy

**Reply by MrsTrellis** [reply]
That's brilliant, thank you Andy.

Is it straightforward to discover the static URL given a URL with a ? ?

**Reply by AndyFx12** [reply]
> 
> 
> That's brilliant, thank you Andy.
> 
> Is it straightforward to discover the static URL given a URL with a ? ?

In most cases, it´s enough to paste URL with ? in your browser - in your testcase, URL changes to static one.

Andy

**Reply by MrsTrellis** [reply]
There was me over-thinking it!!  Thank you, that works perfectly.  I'll check out RadioDroid2 too.

Thanks for all your help.
